### PR TITLE
Drop cockpit.defer() API in favor of async and Promise

### PR DESCRIPTION
**Pull request by martinpitt**
Look ma, no old API!

### CODE REVIEW

File: src/client.js [hunk 1]
```diff
@@ -607,7 +546,60 @@ class RPMOSTreeDBusClient {
             }
         }
 
-        const cleanup = () => {
+        const on_close = (event, ex) => {
+            logDebug("run_transaction", method, method_args, os, ": closed", ex);
+            throw ex;
```
Comment (cockpituous) [inline src/client.js:551]: These 3 added lines are not executed by any test.

File: src/client.js [hunk 2]
```diff
@@ -607,7 +546,60 @@ class RPMOSTreeDBusClient {
...
+        };
+
+        try {
+            await this.connect();
+            const proxy = this.get_os_proxy(os);
+
+            if (!proxy)
+                throw new Error(cockpit.format(_("OS $0 not found"), os));
```
Comment (cockpituous) [inline src/client.js:559]: This added line is not executed by any test.

File: src/client.js [hunk 3]
```diff
@@ -607,7 +546,60 @@ class RPMOSTreeDBusClient {
...
+                            logDebug("run_transaction", method, method_args, os, ": got Finished", args);
+                            if (args) {
+                                if (args[0]) {
+                                    resolve(args[1]);
+                                } else {
+                                    reject(args[1]);
+                                }
+                            } else {
+                                console.warn("Unexpected transaction response", args);
+                                reject(Error({ problem: "protocol-error" }));
```
Comment (cockpituous) [inline src/client.js:596]: These 2 added lines are not executed by any test.

File: src/client.js (hunk 1 above)
Comment (cockpituous) [inline src/client.js:551]: These 3 added lines are not executed by any test.

File: src/client.js (hunk 2 above)
Comment (cockpituous) [inline src/client.js:559]: This added line is not executed by any test.

File: src/client.js (hunk 3 above)
Comment (cockpituous) [inline src/client.js:596]: These 2 added lines are not executed by any test.

File: test/check-ostree [hunk 4]
```diff
@@ -245,6 +245,8 @@ class OstreeRestartCase(testlib.MachineCase):
         m.upload(["files/publickey.asc"], "/root/")
         m.execute("ostree remote gpg-import local -k /root/publickey.asc")
 
+        testlib.sit()
```
Comment (jelly) [inline test/check-ostree:248]: This obviously does not belong in the PR :)

Comment (martinpitt) [reply test/check-ostree:248]: Yup, already resolved locally, I also have some commit message fixups. I'm mostly trying to hunt down [this flake](https://cockpit-logs.us-east-1.linodeobjects.com/pull-560-b8ee88f8-20240327-065534-fedora-coreos/log.html), which at first sight *feels* unrelated, but also is new.

Comment (martinpitt) [reply test/check-ostree]: I added a commit which fixes it for me locally.

File: src/client.js (hunk 1 above)
Comment (cockpituous) [inline src/client.js:551]: These 3 added lines are not executed by any test.

File: src/client.js (hunk 2 above)
Comment (cockpituous) [inline src/client.js:559]: This added line is not executed by any test.

File: src/client.js (hunk 3 above)
Comment (cockpituous) [inline src/client.js:596]: These 2 added lines are not executed by any test.

**Review by jelly — APPROVED** [review]
Amazing! Especially that `notify()` was unused!
